spec/lib/reports/suppliers/pay_your_suppliers_report_spec.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+RSpec.describe "Pay Your Suppliers Report" do
+  let(:hub) { create(:distributor_enterprise) }
+  let(:order_cycle) { create(:open_order_cycle, distributors: [hub]) }
+  let(:product) { order.products.first }
+  let(:variant) { product.variants.first }
+  let(:supplier) { variant.supplier }
+  let(:current_user) { hub.owner }
+  let!(:order) do
+    create(:completed_order_with_totals, distributor: hub, order_cycle:, line_items_count: 1)
+  end
+  let(:params) { { display_summary_row: true } }
+  let(:report) { Reporting::Reports::Suppliers::Base.new(current_user, { q: params }) }
+  let(:report_table_rows) { report.rows }
+
+  context "without fees and taxes" do
+    it "Generates the report" do
+      expect(report_table_rows.length).to eq(1)
+      table_row = report_table_rows.first
+
+      expect(table_row.producer).to eq(supplier.name)
+      expect(table_row.producer_address).to eq(supplier.address.full_address)
+      expect(table_row.producer_abn_acn).to eq("none")
```

Comment:
Interesting, does the report framework automatically convert `nil` to `"none"`?

Reply:
Yes, David. It is 😬

Reply:
Yes, David. It is